### DISCUSSION TITLE
Write errors, except when USB-C cable is connected

### DISCUSSION THREAD

**Discussion by jmbwell** [ZuluSCSI/ZuluSCSI-firmware#457] [Q&A] [answered]
My ZuluSCSI 6.4 board throws write errors when the USB-C cable is disconnected, but works great when the USB-C cable is connected — even if nothing is connected to the other end. Long story short, I feel like there's something electrical happening but I'm not enough of an engineer here to know what.

It was working well when I had it hanging by its SCSI ribbon cable out the expansion port hole in the back of my Mac SE. Today I printed a sled for it and mounted it inside. When I booted up, it started up just fine, but if I tried to save anything to the volume, even just duplicate an existing file, it failed with "A disk error occurred."

I've spent the day trying different SD cards, trying different volume sizes, start sectors, stop sectors, SCSI IDs, termination on and off, all the switches under the General tab in the configuration utility. It would seem like everything I tried worked, but then I'd disconnect USB and put the cover back on the Mac and turn it on, and as soon as I try to duplicate a file, it would fail.

I was getting tired of messing with the case and the guts, and I was settling in to do a lot of troubleshooting, routing a USB-C cable out the back of the case through the security slot. And then it started working.

So I took the cover off, disconnected the cable, power cycled, problem reappeared. Reconnected the cable, power cycled, problem disappeared. I did this back and forth for a while. With the USB-C cable connected, it works. Without the USB-C cable, it doesn't. 

So then I went through all the earlier troubleshooting again. Nothing I had been doing changed anything. Even turning off the termination, it worked with the cable in and did not work with the cable out. 

And by the way, it works great with or without the USB-C cable, if it's hanging outside the enclosure.

Again, I'm not as much of an electrical engineer as I'd like. But I have this hunch this is related to something electrical happening, or something related to EMF radiation inside the enclosure. Whatever the problem is, it seems like maybe the extra impedance or capacitance or something else maybe RF related about the cable is happening, such that when the cable is connected to the USB-C port, the problem is mitigated, but with it disconnected, something is causing just enough of -something- to cause writes and only writes to fail.

Any thoughts on how to proceed to narrow things down, or work around it, happy to hear.

**Comment by aperezbios** [comment, marked as answer]
@jmbwell 
Hm. If you have the ability to do so, can you solder a wire between the two points on the attached image? This will bridge EMI ground and GND, and I suspect it will resolve your issue. Please let me know if it does or does not, and if you do not have the ability to perform this bodge, we can do it for you, if you contact RHC via e-mail, along with your order number.

We believe the underlying issue is that the board firmware is interpreting the read-only pin on the SD card socket as enabled, when in fact it is not.
 
![ZuluSCSI-V6 4-EMIGND-to-GND](https://github.com/user-attachments/assets/c77da5c5-1584-44d0-a863-abe157ad58e0)

**Reply by jmbwell** [reply]
Modification made, problem solved! Thanks so much!

I really appreciate the assist!

**Comment by JaiYenJohn** [comment]
I also had this exact same write problem, only with an Amiga. Same wire fix above worked.
